### PR TITLE
11 get

### PR DESCRIPTION
**Pull request by HalfPint03**

### CODE REVIEW

**Review by NC-Edd-C — COMMENTED** [review]
Looks good!

File: __tests__/app.test.js [hunk 1]
```diff
@@ -270,11 +269,73 @@ describe('GET:200 /api/users', () => {
...
+        });
+        })
+    });
+    test('GET:200 given sort by', () => {
+        return request(app)
+        .get('/api/articles?sort_by=author')
+        .expect(200)
+        .then(({body}) => {
+            expect(body.articles).toBeSortedBy('author', {descending: true})
+        })
```
Comment (nc-mitch-wyatt) [inline __tests__/app.test.js:307]: Great test code! I think the description could be written a little more descriptive, for example here
```
test( 'GET:200 when given a query of sort_by responds with the array of articles sorted by that column'
```

File: __tests__/app.test.js [hunk 2]
```diff
@@ -270,11 +269,73 @@ describe('GET:200 /api/users', () => {
...
+            expect(body.msg).toBe('Bad request')
+        })
+    });
+    test('GET:400 invalid order', () => {
+        return request(app)
+        .get('/api/articles?order=up')
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe('Bad request')
+        })
```
Comment (nc-mitch-wyatt) [inline __tests__/app.test.js:339]: Excellent error testing

File: models/fetch-all-articles.js [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
+    const sortList = ['author', 'title', 'article_id', 'topic', 'created_at', 'votes', 'article_img_url', 'comment_count']
+    const orderList = ['ASC', 'DESC']
+    if (sortList.includes(sort_by) && orderList.includes(order)){
+        return db.query(`SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY articles.${sort_by} ${order}`)
+        .then((articles) => {
+            return articles.rows
+    })
+    } else {
+        return Promise.reject({status: 400, msg: 'Bad request'})
+    }
```
Comment (nc-mitch-wyatt) [inline models/fetch-all-articles.js:13]: I like the way this is written 🏆

**Review by nc-mitch-wyatt — COMMENTED** [review]
Nice work!
I think you could make your test descriptions more verbose but the code looks very nice 👌

**Pull request merged**